### ISSUE TITLE
ofExit() throws error

### ISSUE THREAD

**Issue by K0j0** [openframeworks/openFrameworks#4092]
Hello,

So I pulled the latest from master today and am getting an error whenever I call ofExit() or pres ESC to close a program. I'm seeing this in Linux and OSX. The stack trace ends in ofEvent.h line 194. I can't really tell what's going on there but it seems like the function object it tries to call there is null.

**Comment by bilderbuchi** [comment]
hm, @arturoc I think you dealt with the exit callbacks last, correct?

**Comment by arturoc** [comment]
yes i fixed a couple of issues related to the exit sequence some days ago. it's working fine for me.

is this happening on a any example? or in a project in particular only? can you post the whole stack trace.

**Comment by danoli3** [comment]
I just tested emptyExample on OSX. All is fine. Might be related to @K0j0's project... or outdated fetch of master. Just verify you are on the absolute latest @K0j0.

**Comment by K0j0** [comment]
Oh damn you guys are right. Yeah, a different project just exited fine for me. It's strange though, I'm not overriding the ESC functionality in the project that's giving me an issue. Still, it appears that the problem is definitely in my code and not OF. I'll close this now.

**Comment by arturoc** [comment]
there might be some combination that makes it crash in some projects so even if it's only in your project it might be helpful if you post the stack trace. also in any case the error is surely related with a listener that hasn't been removed from the event before being destroying.

**Comment by K0j0** [comment]
This is after hitting ESC. It looks like functions_copy is empty in the debug window.

![screen shot 2015-07-13 at 11 51 41 am](https://cloud.githubusercontent.com/assets/2118306/8657997/f24f37bc-2955-11e5-9829-1af6a685649a.png)

**Comment by K0j0** [comment]
This is after manually calling ofExit(), which is all that happened in step 3 of this stack trace.

![screen shot 2015-07-13 at 12 08 24 pm](https://cloud.githubusercontent.com/assets/2118306/8658320/ef0cb3f2-2957-11e5-9c9b-7ce919572ec7.png)

**Comment by bakercp** [comment]
When I was dealing with this I had some ofAddListeners that didn't have a matching ofRemoveListeners, leaving dangling event callbacks.

**Comment by arturoc** [comment]
it seems in both cases is crashing when trying to notify the exit event, do you have any object listening to that? that even should be notify before starting the destruction of ofApp but perhaps you are destroying some object before it's exit listener is notified without removing the exit listener?

**Comment by K0j0** [comment]
Thanks for the feedback, guys! Yeah, I definitely wasn't cleaning up a bunch of ofAddListeners but it turns out that that wasn't the problem. Oddly enough, this is getting caused by the ofxUI addon. For some reason deleting a gui from the heap in a destructor (maybe anywhere?) is causing the error.

I haven't updated my version of that addon in awhile so I wonder if a recent pull of OF is no longer compatible with ofxUI? What's even stranger is that I looked up that repo in github and it's apparently been deprecated for like 2 weeks :-(

I'm not sure why that would happen, it was such a great addon. I'm gonna see if I can still get it to work with my project (maybe I'm donig something wrong) but at the moment it seems that the error I saw is caused by the addon no longer working with my project and the latest from OF.
